pandas/core/groupby.py

```diff
@@ -1380,6 +1380,32 @@ def cumsum(self, axis=0, *args, **kwargs):
 
     @Substitution(name='groupby')
     @Appender(_doc_template)
+    def rank(self, axis=0, method='average', numeric_only=True,
+             na_option='keep', ascending=True, pct=False):
+        """Compute numerical data ranks (1 through n) along axis.
+        """
+
+        def wrapper(values):
+            return values.rank(axis=axis, method=method, na_option=na_option,
+                               ascending=ascending, pct=pct)
```

Comment:
I don't think you need to try this first, rather just act on the ``numeric_only`` select FIRST (eg. where you have the ``if numeric_only``, that should be it.